### ISSUE TITLE
Add extra_requires for documentation.

### ISSUE THREAD

**Issue by Julian-O** [websocket-client/websocket-client#734]
In order to build the documentation, the user need to pip install `Sphinx` and `sphinx-rtd-theme`.

This does not appear to be documented anywhere in the project.

These requirement should be recorded as `extra_requires` in setup.py for a `doc` variant.

**Comment by engn33r** [comment]
Good point, these are loosely referenced in the [contributing info](https://github.com/websocket-client/websocket-client/blob/master/CONTRIBUTING.md#i-want-to-contribute-documentation) but probably should appear in setup.py.
